org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/StringTemplateComponent.java

```diff
@@ -0,0 +1,226 @@
+	 */
+	public static final ChildPropertyDescriptor STRING_FRAGMENT_PROPERTY =
+		new ChildPropertyDescriptor(StringTemplateComponent.class, "fragment", StringFragment.class, MANDATORY, CYCLE_RISK); //$NON-NLS-1$
+
+	static {
+		List propertyList = new ArrayList(2);
+		createPropertyList(StringTemplateComponent.class, propertyList);
+		addProperty(EMBEDDED_EXPRESSION_PROPERTY, propertyList);
+		addProperty(STRING_FRAGMENT_PROPERTY, propertyList);
+		PROPERTY_DESCRIPTORS = reapPropertyList(propertyList);
+	}
+
+	StringTemplateComponent(AST ast) {
+		super(ast);
+		supportedOnlyIn21();
+	}
+
+	/**
+	 * The expression;
+	 */
+	private Expression expression = null;
+
+	/**
+	 * The string fragment
+	 */
+	private StringFragment fragment = null;
+
+	/**
+	 * Returns a list of structural property descriptors for this node type.
+	 * Clients must not modify the result.
+	 *
+	 * @param apiLevel the API level; one of the
+	 * <code>AST.JLS*</code> constants
+
+	 * @return a list of property descriptors (element type:
+	 * {@link StructuralPropertyDescriptor})
+	 * @since 3.24
+	 */
+	public static List propertyDescriptors(int apiLevel) {
+		return null;
+	}
+	public static List propertyDescriptors(int apiLevel, boolean previewEnabled) {
+		if (DOMASTUtil.isStringTemplateSupported(apiLevel, previewEnabled)) {
+			return PROPERTY_DESCRIPTORS;
+		}
+		return null;
+	}
+	@Override
+	final List internalStructuralPropertiesForType(int apiLevel) {
+		return propertyDescriptors(apiLevel);
+	}
+
+	@Override
+	final List internalStructuralPropertiesForType(int apiLevel, boolean previewEnabled) {
+		return propertyDescriptors(apiLevel, previewEnabled);
+	}
+
+	@Override
+	final ASTNode internalGetSetChildProperty(ChildPropertyDescriptor property, boolean get, ASTNode child) {
+		if (property == STRING_FRAGMENT_PROPERTY ) {
+			if (get) {
+				return getStringFragment();
+			} else {
+				setStringFragment((StringFragment) child);
+				return null;
+			}
+		}
+		if (property == EMBEDDED_EXPRESSION_PROPERTY ) {
+			if (get) {
+				return getEmbeddedExpression();
+			} else {
+				setEmbeddedExpression((Expression) child);
+				return null;
+			}
+		}
+		// allow default implementation to flag the error
+		return super.internalGetSetChildProperty(property, get, child);
+	}
+	/**
+	 * @noreference This method is not intended to be referenced by clients as it is a part of Java preview feature.
+	 */
+	public void setEmbeddedExpression(Expression processor) {
+		supportedOnlyIn21();
+		if (processor == null) {
+			throw new IllegalArgumentException();
+		}
+		ASTNode oldChild = this.expression;
+		preReplaceChild(oldChild, processor, EMBEDDED_EXPRESSION_PROPERTY);
+		this.expression = processor;
+		postReplaceChild(oldChild, processor, EMBEDDED_EXPRESSION_PROPERTY);
+	}
+
+	/**
+	 * @noreference This method is not intended to be referenced by clients as it is a part of Java preview feature.
```

Comment:
check for the need for @noreference since the class itelf is not to be referenced